### ISSUE TITLE
missing container_memory_rss values after upgrade to Debian 11

### ISSUE THREAD

**Issue by FlyingHavoc** [google/cadvisor#3023]
OS: Debian GNU/Linux 11 (bullseye) arm64
Kernel: 5.10.81-v8+ aarch64
cAdvisor: zcube/cadvisor:latest
Docker:  Server Version: 20.10.11
 Cgroup Driver: systemd
 Cgroup Version: 2
 Runtimes: runc io.containerd.runc.v2 io.containerd.runtime.v1.linux
 containerd version: 7b11cfaabd73bb80907dd23182b9347b4245eb5d
 Security Options:
  apparmor
  seccomp
   Profile: default
  cgroupns
 Docker Root Dir: /var/lib/docker

Summary
After upgrading my Debian 10 to Debian 11 -- my container monitoring is show empty values for `container_memory_rss`

My cadvisor settings are:
` volumes:`
`    - /:/rootfs:ro`
`    - /var/run:/var/run:rw`
`    - /sys:/sys:ro`
`   - /var/lib/docker/:/var/lib/docker:ro`
`    - /dev/disk/:/dev/disk:ro`
`    - /sys/fs/cgroup:/cgroup:ro`
`    command:`
`      - '-housekeeping_interval=10s'`
`      - '-docker_only=true'`
`    restart: unless-stopped`
`    devices:`
`      - /dev/kmsg:/dev/kmsg`

The same settings worked perfectly in Debian 10.
I've upgraded the image to zcube/cadvisor:latest -- this hasn't helped.

Has something changed in Debian 11?
Or am I missing some other cause?
(docker is running as root)

Thanks in advance!

**Comment by Holundermann** [comment]
We are currently evaluating cadvisor to monitor our docker containers and also use debian bullseye as our docker host. We have the exact same behavior for container_memory_rss - cpu usage is also not shown.

On my workstation with ubuntu 20.04 everything works as expected.

**Comment by martadinata666** [comment]
because cgroup v2 enabled debian 11, min ver 0.39

**Comment by Holundermann** [comment]
Hello - just tested - you were right. I used the 'latest' tag which was not updated apparently and now i am using v0.43.0 and all seems to work fine.

**Comment by FlyingHavoc** [comment]
> because cgroup v2 enabled debian 11, min ver 0.39

I'm using ARM64 version, and there is NO docker image for cadvisor and arm64 image. 
Instead I have to use zcube/cadvisor 

I have checked again the docker hub for zcube/cadvisor, but the latest available from them is v0.38.8

Any idea where to get docker image for cadvisor for arm64 and >=0.39?

**Comment by martadinata666** [comment]
the zcube image seems run monthly on schedule from his doc, maybe on February got release 0.39

**Comment by FlyingHavoc** [comment]
> because cgroup v2 enabled debian 11, min ver 0.39

Something strange with the latest cadvisor 0.39.3 now, since I've got it updated from zcube/cadvisor

I see lots of errors and many metrics NOT being collected.
In the cadvisor container logs, I see these error messages flowing all the time:

`W0112 06:04:45.133261       1 container.go:586] Failed to update stats for container "/system.slice/docker-b2373bc3e72d772d51cee51e6801e1f05ce1d919b141a648d036d296038f5dd1.scope": error while statting cgroup v2: [open /sys/kernel/mm/hugepages: no such file or directory
failed to fetch hugetlb info`

`W0112 06:04:49.226834       1 container.go:586] Failed to update stats for container "/system.slice/docker-199b5eee7fd1b80353c727ec781800ca69002e68ed2115959fbd9d571fe02a3b.scope": error while statting cgroup v2: [open /sys/kernel/mm/hugepages: no such file or directory`

`W0112 06:04:51.629646       1 manager.go:696] Error getting data for container /system.slice/docker-c9a41cf04b976fc0892c4c590abd277ac3745ec2c9859231197c68cad9499f78.scope because of race condition`
`W0112 06:04:51.631605       1 manager.go:696] Error getting data for container /system.slice/docker-ff1c00dcefbbef3eeac09f84b877bc61286419a25beb97c876ee497c48d9713e.scope because of race condition`
`W0112 06:04:51.632398       1 manager.go:696] Error getting data for container /system.slice/docker-d81411193030e201d48e8ff6b42191c6cbad8a9c145bca6eb969afae59b43c17.scope because of race condition`
`W0112 06:04:51.633234       1 manager.go:696] Error getting data for container /system.slice/docker-92b28105bed368d3d9aba86308dc8879be68e0b2a695b9d5a0182845068f24d3.scope because of race condition`
`W0112 06:04:51.634035       1 manager.go:696] Error getting data for container /system.slice/docker-d7ba4703dc3b5de5113400be9daeb40eb258233bde5811eba5ccfc3e92d28180.scope because of race condition`
`W0112 06:04:51.634874       1 manager.go:696] Error getting data for container /system.slice/docker-91d7a16dc6513d52ed0bd4c23d243bf13aae0643b34bd41c14fdb89cca8dbe14.scope because of race condition`

**Comment by martadinata666** [comment]
yes i also encountered this, because rpi os don't enable hugetlb by default, and ubuntu does. The latest git got this fix, so another time to wait. Another workaround is stay at 0.38 with cgroup v1 enabled.

**Comment by FlyingHavoc** [comment]
> yes i also encountered this, because rpi os don't enable hugetlb by default, and ubuntu does. The latest git got this fix, so another time to wait. Another workaround is stay at 0.38 with cgroup v1 enabled.

I see this being mentioned in https://github.com/opencontainers/runc/issues/3232
It seems this should be fixed with runc 1.0.3 version.
My runc version is 1.0.2

When checking docker setup with check-config.sh script:
- CONFIG_CGROUP_HUGETLB: missing
- CONFIG_RT_GROUP_SCHED: missing
- CONFIG_SECURITY_SELINUX: missing

What about the errors regarding "Error getting data for container because of race condition"?
`W0112 06:04:51.634035 1 manager.go:696] Error getting data for container /system.slice/docker-d7ba4703dc3b5de5113400be9daeb40eb258233bde5811eba5ccfc3e92d28180.scope because of race condition`

I don't see  them stopping either :/

**Comment by jgramling17** [comment]
For anyone like me who stumbled upon this thread.
The `latest` tag no longer works for `gcr.io/cadvisor/cadvisor`.

You must hardcode it to a newer version like so:
`gcr.io/cadvisor/cadvisor:v0.47.1`

**Comment by Timopheym** [comment]
Thanks a lot @jgramling17! Who could guess that one can break their latest tag...
